modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -1141,86 +1171,57 @@ struct InferList2: public cv::detail::KernelTag {
-
-                            IE::Blob::Ptr blob_0 = extractBlob(*ctx, 0);
-
-                            // Take the next argument, which must be vector (of any kind).
-                            // Use it only to obtain the ROI list size (sizes of all other
-                            // vectors must be equal to this one)
-                            const auto list_size = ctx->inArg<cv::detail::VectorRef>(1u).size();
-                            if (list_size == 0u) {
-                                for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
-                                    auto output = ctx->output(i);
-                                    ctx->out.meta(output, cv::GRunArg::Meta{});
-                                    ctx->out.post(std::move(output));
-                                }
-                                return;
-                            }
-
-                            for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
-                                ctx->outVecR<cv::Mat>(i).resize(list_size);
-                            }
-
-                            // FIXME: This could be done ONCE at graph compile stage!
-                            std::vector< std::vector<int> > cached_dims(ctx->uu.params.num_out);
-                            for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
-                                const IE::DataPtr& ie_out = ctx->uu.outputs.at(ctx->uu.params.output_names[i]);
-                                cached_dims[i] = toCV(ie_out->getTensorDesc().getDims());
-                                // FIXME: Isn't this should be done automatically
-                                // by some resetInternalData(), etc? (Probably at the GExecutor level)
-                                ctx->outVecR<cv::Mat>(i).clear();
-                            }
+        GAPI_Assert(ctx->inArgs().size() > 1u
+                && "This operation must have at least two arguments");
+        IE::Blob::Ptr blob_0 = extractBlob(*ctx, 0);
+        const auto list_size = ctx->inArg<cv::detail::VectorRef>(1u).size();
+        if (list_size == 0u) {
+            for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
+                auto output = ctx->output(i);
+                ctx->out.meta(output, cv::GRunArg::Meta{});
+                ctx->out.post(std::move(output));
+            }
+            return;
+        }
+        // FIXME: This could be done ONCE at graph compile stage!
+        std::vector< std::vector<int> > cached_dims(ctx->uu.params.num_out);
+        for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
+            const IE::DataPtr& ie_out = ctx->uu.outputs.at(ctx->uu.params.output_names[i]);
+            cached_dims[i] = toCV(ie_out->getTensorDesc().getDims());
+            // FIXME: Isn't this should be done automatically
+            // by some resetInternalData(), etc? (Probably at the GExecutor level)
+            ctx->outVecR<cv::Mat>(i).clear();
```

Comment:
this code is also duplicated, how much does it cost to retrieve it to some common place?
Do we have some obstacles and this work seems extra?

Reply:
This isn't my code, so I wouldn't touch it in this at least in this PR, moreover it already has a comment about moving this to common place